Monika-After-Story/MonikaModDev
# Some Derands Lose Affection

This PR expands derandoming and reranding topics. It's now possible to use an override label for derandoming a topic, this allows different dialogue for derandoming topics on a *per-event-basis*. This will override the label used for the label prefix.

Additionally, a `rerandom_callback` has been created which allows us to run functions on rerandom. `args` and `kwargs` are optional to be passed into the data dict, however the function must be callable.

This is currently used to regain half the affection lost by derandoming a bad topic.

## Testing:
- Verify the correct labels are used for derandoming events with and without override derand labels
- Verify topics where Monika vents to you will use the new derandom label
- Verify if a label doesn't exist, we fallback to the label for the label prefix
- Verify rerandom callbacks do not crash and log their reason for failing in `mas_log`.
- Verify no crashes
